src/App.vue

```diff
@@ -1,20 +1,39 @@
 <template>
   <div id="app" class="container">
     <InputBox
-      :todoTitle="todoTitle"
-      :handleAddTodo="handleAddTodo"
-      :handleInput="handleInput"
-      :arrLen="arrLen"
-    />
+      identifier="initalInput"
+      :value="todoTitle"
+      v-model="todoTitle"
+      @handler="handleAddTodo"
+      >Add</InputBox
+    >
 
     <div class="container__todo">
       <div v-for="todo of todos" :key="todo.id" class="container__todo-item">
-        <TodoTitle :done="todo.done" :title="todo.title" />
-
-        <TodoButton v-if="!todo.done" :handler="() => handleDone(todo.id)"
-          >Done</TodoButton
+        <TodoTitle
+          v-if="!todo.inEditState"
+          :done="todo.done"
+          :title="todo.title"
+        />
+
+        <InputBox
+          v-else
+          identifier="todoInput"
+          :todo="todo"
+          :id="todo.id"
+          :value="todo.title"
+          v-model="todo.title"
+          @handler="handleUpdate"
+          @handleCancel="handleCancel"
+          >Update</InputBox
```

Comment:
Please use InputBox component for only input field purposes. Bring out the buttons.